packages/backend/src/auth/lightdashJwt.ts

```diff
@@ -0,0 +1,89 @@
+import {
+    CreateEmbedJwt,
+    EmbedJwt,
+    EmbedJwtSchema,
+    ForbiddenError,
+    getErrorMessage,
+    isDashboardUuidContent,
+    ParameterError,
+} from '@lightdash/common';
+import * as Sentry from '@sentry/node';
+import jwt from 'jsonwebtoken';
+import { z } from 'zod';
+import { lightdashConfig } from '../config/lightdashConfig';
+import Logger from '../logging/logger';
+import { EncryptionUtil } from '../utils/EncryptionUtil/EncryptionUtil';
+
+export const JWT_HEADER_NAME = 'lightdash-embed-token';
+
+/**
+ * Encodes JWT data into a token
+ */
+export function encodeLightdashJwt(
+    jwtData: CreateEmbedJwt,
+    encodedSecret: string | Buffer,
+    expiresIn: string,
+): string {
+    const encryptionUtil = new EncryptionUtil({ lightdashConfig });
+    const secret = encryptionUtil.decrypt(
+        Buffer.isBuffer(encodedSecret)
+            ? encodedSecret
+            : Buffer.from(encodedSecret),
+    );
+    return jwt.sign(jwtData, secret, { expiresIn });
+}
+
+/**
+ * Decodes and validates a Lightdash Embed JWT token
+ */
+export function decodeLightdashJwt(
+    token: string,
+    encodedSecret: string | Buffer,
+): EmbedJwt {
+    try {
+        const encryptionUtil = new EncryptionUtil({ lightdashConfig });
+        const secret = encryptionUtil.decrypt(
+            Buffer.isBuffer(encodedSecret)
+                ? encodedSecret
+                : Buffer.from(encodedSecret),
+        );
+        const decodedToken = jwt.verify(token, secret) as EmbedJwt;
+
+        // Alert if the token is not in the expected format so we can inform the org before enforcing validation
+        try {
+            // Not returning the inferred zod type because tsoa doesn't support it
+            EmbedJwtSchema.parse(decodedToken);
+        } catch (e) {
+            let errorIdentifier = 'unknown';
+            if (decodedToken?.content) {
+                if (isDashboardUuidContent(decodedToken.content)) {
+                    errorIdentifier = decodedToken.content.dashboardUuid;
+                } else if ('dashboardSlug' in decodedToken.content) {
+                    errorIdentifier = decodedToken.content.dashboardSlug;
+                }
+            }
+
+            let errors: string;
+            if (e instanceof z.ZodError) {
+                const zodErrors = e.issues
+                    .map((issue) => issue.message)
+                    .join(', ');
+
+                errors = zodErrors;
+            } else {
+                errors = getErrorMessage(e);
+            }
+
+            throw new ParameterError(`Invalid embed token: ${errors}`);
```

Comment:
@owlas I cleaned this up a bit from the extracted service reading the earlier graphite error. We shouldn't need to throw, log, and emit to Sentry given the common express error handler. I would imagine we were double logging everything if it ever occurred.